### PR TITLE
New resource and various bug fixes

### LINKED ISSUE #493: Add SPSearchCrawlStore resource
**Details of the scenario you try and problem that is occurring:**
For a large search topology I'm missing a few things. One is that all the databases are created in the same sql instance. The name cannot be configured individually and there is only one property for the sql server. And I need to create multiple crawl store databases which is not possible with the current version. The multiple crawl databases are also the reason why it would be nice to name the first crawl store db ...crawlstore01.
Could this be implemented simular to the index partitions with an additional resource like SPSearchCrawlStore to create additional crawl dbs? 

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Windows 2012, PS5

**Version of the DSC module you're using:**
1.5.0.0

### LINKED ISSUE #1389: [SPShellAdmins] Setting specific DBs did not succeed
### Problem description

I always get an error when setting the ShellAdmins. This occurs on the BCS-DB and Secure Store.
As far as I noticed the DBs are created differently than the others. In the SQL the farm account is not set at Security > Users at the specific DBs. Instead the dbo-Account uses the schema of the farm account. Could that be the problem?

### Verbose logs

```text
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
     "Cannot add DEV\Farm to the SharePoint_Shell_Access role of the database SP16-JUMP_ServiceApplication_BusinessDataConnectivity. A possible cause of this error
is that the account name was already added to the database as a login using a different user name than the account name."
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Get-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving BeginProcessing Method of Add-SPShellAdmin.
Some or all identity references could not be translated.
    + CategoryInfo          : InvalidData: (Microsoft.Share...AddSPShellAdmin:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletAddSPShellAdmin
    + PSComputerName        : SERVER01

VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving ProcessRecord Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]:                            [[SPShellAdmins]AddShellAdmins] Leaving EndProcessing Method of Add-SPShellAdmin.
VERBOSE: [SERVER01]: LCM:  [ End    Set      ]  [[SPShellAdmins]AddShellAdmins]  in 0.8280 seconds.
The PowerShell DSC resource '[SPShellAdmins]AddShellAdmins' with SourceInfo 'D:\INSTALL\DSC\SharePoint.ps1::837::17::SPShellAdmins' threw one or more non-terminating
errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for
more details.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : NonTerminatingErrorFromProvider
    + PSComputerName        : SERVER01
```

### DSC configuration

```powershell
SPShellAdmins AddShellAdmins {
                    MembersToInclude     = @("DEV\Farm","DEV\Admins","DEV\ShellAdmins")
                    AllDatabases         = $true
                    IsSingleInstance     = "Yes"
                    PsDscRunAsCredential = $SpFarmAccount
                    DependsOn            = "[SPFarm]CreateOrJoinSharePointFarm"
                }
```

### Suggested solution

Actually I don't know whether my description includes the solution?!

### SharePoint version and build

```text
SharePoint 2016
16.0.5227.1000
```

### Operating system the target node is running

```text
OsName               : Microsoft Windows Server 2016 Standard
OsOperatingSystemSKU : StandardServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.4825.amd64fre.rs1_release.211202-1611
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
```

### PowerShell version and build the target node is running

```text
Name                           Value
----                           -----
PSVersion                      5.1.14393.4583
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.4583
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### SharePointDsc version

```text
Name          Version Path
----          ------- ----
SharePointDSC 4.5.1   C:\Program Files\WindowsPowerShell\Modules\SharePointDSC\4.5.1\SharePointDSC.psd1
```

### LINKED ISSUE #1404: [SPSearchIndexPartition] Regression introduced with with the fix for issue #1376
### Problem description

The fix for issue #1376 unfortunately introduced a regression issue:
https://github.com/dsccommunity/SharePointDsc/blob/df57d18ba35b26017788ab569cece24af4aa57d4/SharePointDsc/DSCResources/MSFT_SPSearchIndexPartition/MSFT_SPSearchIndexPartition.psm1#L34-L41

The added `Select-Object -First 1` only selected one Index Component. This meant that the Get method only returned one index partition (line 40), instead of only returning a single Root Directory.

### Verbose logs

```text
2022-04-22T11:48:41.4910790Z VERBOSE: [SEARCH1]:                            [[SPSearchIndexPartition]IndexPartition1] Current Values: 
2022-04-22T11:48:41.4923044Z Index=1; RootDirectory=L:\SPIndex\1; Servers=SEARCH3; ServiceAppName=Search SA
2022-04-22T11:48:41.4936309Z VERBOSE: [SEARCH1]:                            [[SPSearchIndexPartition]IndexPartition1] Target Values: Index=1;
2022-04-22T11:48:41.4949400Z  RootDirectory=L:\SPIndex\1; Servers=(SEARCH2,SEARCH3); ServiceAppName=Search SA; Verbose=True
```

### DSC configuration

```powershell
Any configuration with two servers running an index component
```

### Suggested solution

```PowerShell
        $searchComponent = Get-SPEnterpriseSearchComponent -SearchTopology $currentTopology -Verbose:$false | `
                Where-Object -FilterScript {
                ($_.GetType().Name -eq "IndexComponent") `
                    -and ($_.IndexPartitionOrdinal -eq $params.Index)
            }

        $IndexComponents = $searchComponent.ServerName
        $rootDirectory = ($searchComponent | Select-Object -First 1).RootDirectory
```

### SharePoint version and build

```text
Any
```

### Operating system the target node is running

```text
Any
```

### PowerShell version and build the target node is running

```text
Any
```

### SharePointDsc version

```text
v5.1
```

### PR DESCRIPTION
**Pull request by ykuijs**
#### Pull Request (PR) description

This PR adds the SPSearchCrawlDatabase resource, adds extra troubleshooting information, implements export logic for SPShellAdmin, increases export requirement to ReverseDsc v2.0.0.11 and fixes several bugs.

#### This Pull Request (PR) fixes the following issues

- Fixes #493 
- Fixes #1404 
- Fixes #1389 

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof and comment-based
      help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1406)

### CODE REVIEW

**Comment by ykuijs** [comment]
@NikCharlebois Up for a review?

File: SharePointDsc/DSCResources/MSFT_SPSearchCrawlDatabase/MSFT_SPSearchCrawlDatabase.schema.mof [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[ClassVersion("1.0.0.0"), FriendlyName("SPSearchCrawlDatabase")]
+class MSFT_SPSearchCrawlDatabase : OMI_BaseResource
+{
+    [Key, Description("The name of the crawl database")] string DatabaseName;
+    [Key, Description("The name of the search service application")] string ServiceAppName;
+    [Write, Description("The server that should host the crawl databases")] string DatabaseServer;
+    [Write, Description("Should SQL Server authentication be used to connect to the database?")] Boolean UseSQLAuthentication;
+    [Write, Description("If using SQL authentication, the SQL credentials to use to connect to the instance"), EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
+    [Write, Description("Present if the service app should exist, absent if it should not"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
```
Comment (NikCharlebois) [inline SharePointDsc/DSCResources/MSFT_SPSearchCrawlDatabase/MSFT_SPSearchCrawlDatabase.schema.mof:9]: In this case it doesn't deal with the Service App, but rather with the crawl db. We should update the definition

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Need to update some of the descriptions.

**Review by NikCharlebois — APPROVED** [review]
LGTM

**Pull request merged**